cmd/root.go

```diff
@@ -287,24 +292,17 @@ func parseConfig(cmd *cobra.Command, conf *proxy.Config, args []string) error {
 				ic.UnixSocket = u[0]
 			}
 
-			if iam, ok := q["auto-iam-authn"]; ok {
-				if len(iam) != 1 {
-					return newBadCommandError(fmt.Sprintf("auto iam authn param should be only one value: %q", iam))
-				}
-				switch iam[0] {
-				case "true", "t":
-					enable := true
-					ic.IAMAuthN = &enable
-				case "false", "f":
-					disable := false
-					ic.IAMAuthN = &disable
-				default:
-					return newBadCommandError(
-						fmt.Sprintf("auto iam authn query param should be true or false, got: %q",
-							iam[0],
-						))
-				}
+			iam, err := parseBoolOpt("auto-iam-authn", q)
+			if err != nil {
+				return err
 			}
+			ic.IAMAuthN = iam
+
+			privateIP, err := parseBoolOpt("private-ip", q)
+			if err != nil {
+				return err
+			}
+			ic.PrivateIP = privateIP
```

Comment:
any reason not to just assign these directly to the config?

Reply:
done